pkg/api/types.go

```diff
@@ -628,6 +628,8 @@ type LiteralTestStep struct {
 	// Dependencies lists images which must be available before the test runs
 	// and the environment variables which are used to expose their pull specs.
 	Dependencies []StepDependency `json:"dependencies,omitempty"`
+	// DnsConfig for step's Pod.
+	DNSConfig StepDNSConfig `json:"dnsConfig,omitempty"`
```

Comment:
Lets make all the DNSConfig fields pointers to avoid serializing it into all config